### DISCUSSION TITLE
Ability to scan music within a rar/zip archive

### DISCUSSION THREAD

**Discussion by hrija** [navidrome/navidrome#3252] [Ideas]
I think it'd be nice if Navidrome could scan rar/zip files for songs, seeing as I've come across a fair few instances where people like to keep their music organized in archives for whatever reason. 

Is implementing something like this a lot more complicated than it seems?

Would it be feasible at all to have archives scanned too during library scans - adding song info to the database, and the actual music being unpacked on demand?

I've heard foobar has similar functionality with its foo_unpack module, so it'd be interesting to see if Navidrome could implement something akin to this.

**Comment by norohind** [comment]
As a workaround, I'm pretty sure you can mount an archive and point navidrome to the mount point.

**Reply by hrija** [reply]
Yeah that's true. I just gave rar2fs a try and I'm stunned by how cleanly it's mounted everything, and how well it seems to be working with Navidrome. If anyone stumbles upon this in the future give it a try :D
